level.go

```diff
@@ -101,7 +105,10 @@ func (l *moduleLeveled) GetLevel(module string) Level {
 
 // SetLevel sets the log level for the given module.
 func (l *moduleLeveled) SetLevel(level Level, module string) {
-	l.levels[module] = level
+	levels := l.levels.Load().(moduleLeveledMap)
+	l.lock.Lock()
+	levels[module] = level
+	l.lock.Unlock()
```

Comment:
I could be wrong, but looks like the `levels` map is being modified, yet the reader in `GetLevel` is not underneath a lock so this can race the read and the write.

Maybe there's a way to even get rid of the map lookup in regular logging. Maybe GetLevel needs to return a pointer or sync.Value. And the IsEnabledFor can cache that (it is never removed)